### PR TITLE
Add client-go in the accessing cluster doc

### PR DESCRIPTION
**Pull request by caesarxuchao**
ref: https://github.com/kubernetes/kubernetes/pull/31751

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/kubernetes.github.io/1127)

### CODE REVIEW

File: docs/user-guide/accessing-the-cluster.md [hunk 1]
```diff
@@ -100,13 +100,36 @@ with future high-availability support.
...
-There are [client libraries](https://github.com/kubernetes/kubernetes/tree/{{page.githubbranch}}/docs/devel/client-libraries.md) for accessing the API
-from several languages.  The Kubernetes project-supported
-[Go](http://releases.k8s.io/{{page.githubbranch}}/pkg/client/)
-client library can use the same [kubeconfig file](/docs/user-guide/kubeconfig-file)
-as the kubectl CLI does to locate and authenticate to the apiserver.
+The Kubernetes project-supported Go client library is at [https://github.com/kubernetes/client-go](https://github.com/kubernetes/client-go).
+
+To use it, 
+* Run `go get k8s.io/client-go/1.4/kubernetes` to get the library. 1.4 is a release version of your choice.
+* Write application atop of the client-go clients. Note that client-go defines its own API objects, so if needed, please import API definitions from client-go rather than from the main repository, e.g., `import "k8s.io/client-go/1.4/pkg/api"` is correct.
```
Comment (lavalamp) [inline docs/user-guide/accessing-the-cluster.md]: v1 ;)

File: docs/user-guide/accessing-the-cluster.md [hunk 2]
```diff
@@ -100,13 +100,36 @@ with future high-availability support.
 
 ### Programmatic access to the API
 
-There are [client libraries](https://github.com/kubernetes/kubernetes/tree/{{page.githubbranch}}/docs/devel/client-libraries.md) for accessing the API
```
Comment (lavalamp) [inline docs/user-guide/accessing-the-cluster.md]: probably shouldn't lose this link.

**Comment by lavalamp** [comment]
looks great as soon as we get the example files in.

**Comment by nikhiljindal** [comment]
Did you want to send this PR to release-1.4 branch?

**Comment by caesarxuchao** [comment]
Yes, we want to. Do you know what milestone to apply? `launchday`?

File: docs/user-guide/accessing-the-cluster.md (hunk 2 above)
Comment (caesarxuchao) [inline docs/user-guide/accessing-the-cluster.md]: It's not lost, it's moved to after the example.

**Comment by caesarxuchao** [comment]
@lavalamp can I get the LGTM? Thanks.

**Comment by lavalamp** [comment]
@devin-donnelly Is this OK to merge in the master branch or should it go to the 1.4 branch?

**Comment by caesarxuchao** [comment]
I think merging in master is ok, there is nothing specific to 1.4 here.

**Comment by devin-donnelly** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[docs/user-guide/accessing-the-cluster.md, line 106 \[r2\]](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127#-KRG1_qxr-sv3vIZWscb:-KRG1_qxr-sv3vIZWscc:b-oanheq) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/7d6d512925e3bc2d12d83d169e487f043f1a6b0a/docs/user-guide/accessing-the-cluster.md#L106)):*
> ```Markdown
> 
> To use it, 
> * Run `go get k8s.io/client-go/1.4/kubernetes` to get the library. 1.4 is a release version of your choice.
> ```

I would mention that 1.4 is the latest version, and tell them what the minimum supported version is.

---

*[docs/user-guide/accessing-the-cluster.md, line 107 \[r2\]](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127#-KRG1nMcNgLFAnaXX_sD:-KRG1nMcNgLFAnaXX_sE:b-spf73d) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/7d6d512925e3bc2d12d83d169e487f043f1a6b0a/docs/user-guide/accessing-the-cluster.md#L107)):*
> ```Markdown
> To use it, 
> * Run `go get k8s.io/client-go/1.4/kubernetes` to get the library. 1.4 is a release version of your choice.
> * Write application atop of the client-go clients. Note that client-go defines its own API objects, so if needed, please import API definitions from client-go rather than from the main repository, e.g., `import "k8s.io/client-go/1.4/pkg/api/v1"` is correct.
> ```

Write application -> Write an application

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127)*

**Comment by devin-donnelly** [comment]
*[docs/user-guide/accessing-the-cluster.md, line 107 \[r1\]](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127#-KQSd3LY52ZLQU6yqYcb-r1-107:-KRG3fK08_3cMOvawb1w:bunrpo3) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/ba31cd124f5a734ba24c3c9f25ec0e976d60e471/docs/user-guide/accessing-the-cluster.md#L107)):*
<details><summary><i>Previously, lavalamp (Daniel Smith) wrote…</i></summary>
> v1 ;)
</details>
Right, but we don't want the user to have to guess what exact number to type in. 1.4 works, so I assume it's also 1.0, 1.1, 1.2, 1.3, etc.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127)*

**Comment by caesarxuchao** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[docs/user-guide/accessing-the-cluster.md, line 106 \[r2\]](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127#-KRG1_qxr-sv3vIZWscb:-KRG9ADi4EwUDt6yqaEE:b-3rveqw) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/7d6d512925e3bc2d12d83d169e487f043f1a6b0a/docs/user-guide/accessing-the-cluster.md#L106)):*
<details><summary><i>Previously, devin-donnelly wrote…</i></summary>
> I would mention that 1.4 is the latest version, and tell them what the minimum supported version is.
</details>
How about "1.4 is a release version of your choice. Visit https://github.com/kubernetes/client-go to see what versions are supported"?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127)*

**Comment by caesarxuchao** [comment]
@devin-donnelly PTAL. Thanks.

**Comment by devin-donnelly** [comment]
*[docs/user-guide/accessing-the-cluster.md, line 106 at r2](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127#-KRG1_qxr-sv3vIZWscb:-KRf1yttYW5GKdYTFnKo:b-m3y4nn) ([raw file](https://github.com/kubernetes/kubernetes.github.io/blob/7d6d512925e3bc2d12d83d169e487f043f1a6b0a/docs/user-guide/accessing-the-cluster.md#L106)):*
<details><summary><i>Previously, caesarxuchao (Chao Xu) wrote…</i></summary>
> How about "1.4 is a release version of your choice. Visit https://github.com/kubernetes/client-go to see what versions are supported"?
</details>
I would change to:

"To get the library, run the following command: `go get k8s.io/client-go/<version number>/kubernetes` See [https://github.com/kubernetes/client-go](https://github.com/kubernetes/client-go) to see which versions are supported."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/kubernetes.github.io/1127)*

**Comment by devin-donnelly** [comment]
@caesarxuchao One suggested revision.

**Comment by caesarxuchao** [comment]
@devin-donnelly thanks for the edits. PTAL. Thanks.

**Pull request merged**